src/compiler/scala/tools/nsc/settings/ScalaSettings.scala

```diff
@@ -133,7 +133,8 @@ trait ScalaSettings extends StandardScalaSettings with Warnings {
   val mainClass          = StringSetting       ("-Xmain-class", "path", "Class for manifest's Main-Class entry (only useful with -d <jar>)", "")
   val sourceReader       = StringSetting       ("-Xsource-reader", "classname", "Specify a custom method for reading source files.", "")
   val reporter           = StringSetting       ("-Xreporter", "classname", "Specify a custom subclass of FilteringReporter for compiler messages.", "scala.tools.nsc.reporters.ConsoleReporter")
-  val source             = ScalaVersionSetting ("-Xsource", "version", "Treat compiler input as Scala source for the specified version, see scala/bug#8126.", initial = ScalaVersion("2.13"))
+  val source             = ScalaVersionSetting ("-Xsource", "version", "Enable features that will be available in a future version of Scala.", initial = version213).withPostSetHook(s =>
+      if (s.value < version213) errorFn.apply(s"-Xsource must be at least the current major version (${version213.versionString})"))
```

Comment:
At some point, it was the opposite: Scala IDE with 2.11 with `-Xsource:2.10` would make the presentation compiler behave "more like" 2.10.

With this use case gone, I think it's fine to restrict `-Xsource` to the future.

Reply:
Under `-Xsource:3` we'll be able to use `-source` for `3.0` support when running `3.1`.

Reply:
Instead of smileys, I'll just add laugh emoji to my own joke, so it's clear. I realize it is not in good taste.